File: factorial-recursion/java/FactorialRecursion.java
import java.util.Scanner;

public class FactorialRecursion {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number : ");
        int input = scanner.nextInt();

        if (input < 0) {
            System.out.println("Invalid user input. Please enter a positive number.");
            scanner.close();
            return;
        }

        int userInput = input;
        long factorial = calculateFactorial(userInput);

        System.out.println("Factorial of " + userInput + " is " + factorial);

        scanner.close();
    }

    static long calculateFactorial(int number) {
        if (number == 0) {
            return 1;
        } else if (number == 1) {
            return number;
        } else {
            return calculateFactorial(number - 1) * number;
        }
    }

}

File: factorial-recursion/python/FactorialRecursion.py
def main():
    print('Enter a number: ')
    user_input = int(input());

    if user_input < 0:
        print('Invalid user input. Please enter a positive number.')
        return

    original_input = int(user_input) 
    factorial = get_factorial(original_input)

    print('Factorial of ' + str(original_input) + ' is ' + str(factorial))

# Factorial of a number using recursion
def get_factorial(n):
   if n == 1:
       return n
   else:
       return n * get_factorial(n-1)


if __name__ == '__main__':
    main()